src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -354,7 +354,7 @@ private void initMappers(Map<String, Object> withoutType) {
                     path = fieldNode.toString();
                 } else if (fieldName.equals("format")) {
                     format = fieldNode.toString();
-                } else if (fieldName.equals("default")) {
+                } else if (fieldName.equals("default") && fieldNode != null) {
```

Comment:
Isn't the same NPE possible above in all the other conditions?  Should we just continue the loop before these checks?

```
if (fieldNode == null) continue;
```